src/web/extractors.rs

```diff
@@ -88,11 +91,39 @@ impl BatchBsoBody {
     }
 }
 
-fn get_trimmed_header(headers: &HeaderMap, key: HeaderName, default: &HeaderValue) -> String {
-    let ct_raw = std::str::from_utf8(headers.get(key).unwrap_or(&default).as_bytes())
+// This tries to do the right thing to get the Accepted header according to
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept, but some corners can absolutely be cut.
+// This will pull the first accepted content type listed, or the highest rated non-accepted type.
+fn get_weighted_header(
+    headers: &HeaderMap,
+    key: HeaderName,
+    accepted: &[&str],
+    default: &'static str,
+) -> String {
+    let def_hv = HeaderValue::from_static(default);
+    let hv_raw = std::str::from_utf8(headers.get(key).unwrap_or(&def_hv).as_bytes())
         .unwrap_or_else(|_| "invalid");
-    let ct_parts: Vec<&str> = ct_raw.split(';').collect();
-    ct_parts[0].trim_end().to_owned()
+    let hv_parts: Vec<&str> = hv_raw.split(',').collect();
+    let mut choice_weight = 0;
+    let mut pick: String = default.to_owned();
+    for choice in hv_parts {
+        let opt_weight: Vec<&str> = choice.split(';').collect();
+        let opt = opt_weight[0].trim_end().to_lowercase();
+        if accepted.contains(&&opt.as_ref()) {
+            return opt;
+        }
+        let weight = if opt_weight.len() > 1 {
+            let weights: Vec<&str> = opt_weight[1].split('=').collect();
+            i64::from_str(weights[1]).unwrap_or_else(|_| 0)
```

Comment:
weights.len() could be 1 here

nit: if this still exists, could use unwrap_or(0) instead? (or am I wrong, I thought clippy caught this?)

Reply:
clippy usually wants the `unwrap_or_else()` rather than `unwrap_or()` so I defaulted to that.

Reply:
only when the or arg is some kind of fn call, I guess it just doesn't notice the inverse